src/main/scala/com/microsoft/azure/iot/iothubreact/scaladsl/IoTHub.scala

```diff
@@ -70,6 +73,11 @@ class IoTHub(config: IConfiguration)
   def propertySink: Sink[DeviceProperties, Future[Done]] =
     DevicePropertiesSink(config).scalaSink()
 
+  /**
+    * Provides an offset sink that can be incorporated into a graph for at-least-once semantics (withCheckpoints should be false)
+    */
```

Comment:
The equivalent in the new API is `saveOffsetsOnPull`, which is part of `SourceOptions`, which is only available to the `source(..)` method. If that went into the config that was sent to IoTHub, this check could be made. But that seems out of keeping with the purpose of the configuration object, which is referring to connectivity but not behavior. Perhaps it's good enough that we've named that option `saveOffsetsOnPull` such that a developer with reasonable understanding of what they're doing can understand why both should not be used at once?